### PR TITLE
login, restore, homescreen pages, routes

### PR DESCRIPTION
**Pull request by 8hokage**
Please @tiero review

Markup of pages: Setup wallet, Homescreen, restore wallet without any logic.
also added icons and rows.

### CODE REVIEW

**Comment by tiero** [comment]
Please add instructions in the Readme on how to get up and running useful for other developers. [Like this for example](https://github.com/tdex-network/tdex-sdk#-local-development) I woudl add them in the `scripts` field in the `package.json` and reference them.

File: package.json [hunk 1]
```diff
@@ -3,25 +3,38 @@
...
+    "@capacitor/android": "^2.4.2",
+    "@capacitor/core": "2.4.2",
+    "@capacitor/ios": "^2.4.2",
+    "@ionic-native/barcode-scanner": "^5.29.0",
     "@ionic/react-hooks": "0.0.8",
-    "@ionic/react-router": "^5.0.7",
+    "@ionic-native/core": "^5.29.0",
+    "@ionic-native/screen-orientation": "^5.29.0",
+    "@ionic/react": "^5.0.7",
+    "ionicons": "^5.0.0",
```
Comment (tiero) [inline package.json:15]: duplicate

File: package.json [hunk 2]
```diff
@@ -3,25 +3,38 @@
...
     "react-router": "^5.1.2",
     "react-router-dom": "^5.1.2",
+    "classnames": "^2.2.6",
+    "cordova-plugin-screen-orientation": "^3.0.2",
+    "es6-promise-plugin": "^4.2.2",
+    "ionicons": "^5.2.3",
+    "phonegap-plugin-barcodescanner": "^8.1.0",
+    "react": "^16.13.0",
+    "react-dom": "^16.13.0",
+    "react-router": "^5.2.0",
```
Comment (tiero) [inline package.json:36]: duplicate

File: package.json [hunk 3]
```diff
@@ -3,25 +3,38 @@
...
+    "react-router-dom": "^5.2.0",
```
Comment (tiero) [inline package.json:37]: duplicate

**Review by tiero — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -3,25 +3,38 @@
...
+    "@ionic-native/core": "^5.29.0",
+    "@ionic-native/screen-orientation": "^5.29.0",
+    "@ionic/react": "^5.0.7",
+    "ionicons": "^5.0.0",
+    "@ionic/react-router": "^5.4.1",
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.4.0",
     "@testing-library/user-event": "^8.0.3",
+    "@types/classnames": "^2.2.11",
     "@types/jest": "^24.0.25",
```
Comment (tiero) [inline package.json:21]: @types could go into devDependencies, isn'it?

File: src/pages/Homescreen/index.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import './style.scss';
+
+const Homescreen: React.FC<RouteComponentProps> = ({history}) => {
+  return (
+    <IonPage>
+    <div className="gradient-background"></div>
+      <IonContent>
+        <div className="main-page-wrapper">
+          <img className="logo" src="./assets/img/logo.png" />
```
Comment (tiero) [inline src/pages/Homescreen/index.tsx:13]: missing `alt` prop

File: src/theme/global.scss [hunk 6]
```diff
@@ -0,0 +1,112 @@
+@import "./variables.scss";
+@import "./components/exchange-coin-block.scss";
```
Comment (tiero) [inline src/theme/global.scss:2]: this doesen't exists

**Review by tiero — CHANGES_REQUESTED** [review]

File: src/theme/global.scss (hunk 6 above)
Comment (8hokage) [reply src/theme/global.scss:2]: Yep, i tried split code to different blocks, mb forgot smthng

Comment (tiero) [reply src/theme/global.scss:2]: Otherwise the project fails to start

Comment (tiero) [reply src/theme/global.scss:2]: @8hokage searc.scss doesent exists neither

File: README.md [hunk 7]
```diff
@@ -1,2 +1,10 @@
 # tdex-app
 📱 Mobile app for making trades on TDEX 
+
+## Run
+
+1. Clone repo
+
+2. ```sh  $ npm install```
+
+3. ```sh $ npm run start``` it opens it in browser
```
Comment (tiero) [inline README.md:10]: It's missing how to create a build. (wich on my computer fails with 

```
./src/App.tsx
Cannot find file './pages/Tabs' in './src'.

```

Are you sure we dont' need to inform the developer how to build the scss too?

Comment (8hokage) [reply README.md:10]: it fails cause i think i need add all scss, my bad

Comment (8hokage) [reply README.md:10]: @tiero I think now it should work, it was bad idea maybe to not split code for small prs while i was doing markup

File: src/routes.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    path: routerLinks.tradeSummary,
+    component: TradeSummary,
+  },
+  { 
+    path: routerLinks.withdrawal,
+    component: Withdrawal,
+  },
+  {
+    path: routerLinks.recieve,
+    component: Recieve,
```
Comment (tiero) [inline src/routes.ts:51]: typo: It should be Receive

File: src/routes.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
+import { IconExchange, IconSettings, IconWallet } from "./components/icons"
+import Exchange from "./pages/Exchange"
```
Comment (tiero) [inline src/routes.ts:2]: It fails to build 

```
import Exchange from "./pages/Exchange"
```

**Comment by tiero** [comment]
Still have errors. 

```
Unhandled Rejection (NotSupportedError): Operation is not supported
```

I think it's pointless  to make a review. if we do not have a stable way to reproduce the build first.

Please try it first and ping me once its more stable.

File: src/App.tsx [hunk 10]
```diff
@@ -34,7 +34,12 @@ const App: React.FC = () => {
     const { StatusBar } = Plugins;
     StatusBar.setBackgroundColor({color: "#333333"});
 
-    ScreenOrientation.lock(ScreenOrientation.ORIENTATIONS.PORTRAIT);
+    try {
+      ScreenOrientation.lock(ScreenOrientation.ORIENTATIONS.PORTRAIT);
+    } catch (err) {
+      console.log(err);
```
Comment (tiero) [inline src/App.tsx:40]: Did you tested?

Comment (tiero) [reply src/App.tsx:40]: It cannot work. isn't it inside an async await. So you should resolve the returning promise in the normal whay then/catch

**Pull request merged**